pkg/cri/resource-manager/requests.go

```diff
@@ -51,6 +51,75 @@ func (m *resmgr) setupRequestProcessing() error {
 	return nil
 }
 
+// disambiguate produces disambiguation context for a request/reply dump.
+func (m *resmgr) disambiguate(msg interface{}) string {
+	var qualifier string
+
+	m.RLock()
+	defer m.RUnlock()
+
+	switch req := msg.(type) {
+	case *criapi.RunPodSandboxRequest:
+		if req.Config != nil && req.Config.Metadata != nil {
+			qualifier = req.Config.Metadata.Name
+		}
+	case *criapi.StopPodSandboxRequest:
+		if pod, ok := m.cache.LookupPod(req.PodSandboxId); ok {
+			qualifier = pod.GetName()
+		} else {
+			qualifier = "unknown pod " + req.PodSandboxId
+		}
+	case *criapi.RemovePodSandboxRequest:
+		if pod, ok := m.cache.LookupPod(req.PodSandboxId); ok {
+			qualifier = pod.GetName()
+		} else {
+			qualifier = "unknown pod " + req.PodSandboxId
+		}
```

Comment:
Could these be combined with `fallthrough`(?)

Reply:
At least I can't see how. At least not in a way that'd give reasonable reduction of (syntactic) code duplication. And the reason I don't see a way is the type switch. For this most obvious try:
```
	switch req := msg.(type) {
...
	case *criapi.StopPodSandboxRequest,	*criapi.RemovePodSandboxRequest:
		if pod, ok := m.cache.LookupPod(req.PodSandboxId); ok {
			qualifier = pod.GetName()
		} else {
			qualifier = "unknown pod " + req.PodSandboxId
		}
		if pod, ok := m.cache.LookupPod(req.PodSandboxId); ok {
			qualifier = pod.GetName()
		} else {
			qualifier = "unknown pod " + req.PodSandboxId
		}
```
the compiler seems to generate code where `req` in that combined case has the generic `interface{}` type, so derefering fields don't work. You'd need to switch on the type again which destroys the whole point.. Golang does not have real duck-typing even if it syntactically might occasionally seem so.

But of course I might miss something mode obvious here...

Reply:
Ach, you're correct. I was reading through the code too quickly.